### PR TITLE
Allow to pass native endpoint options to index creation

### PR DESCRIPTION
**Pull request by deguif**
This PR will allow to pass endpoint query parameters to index creation: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-query-params

The list of available options is retrieved from the corresponding elasticsearch endpoint class to avoid maintaining it here.

### CODE REVIEW

File: src/Index.php
Comment (ruflin) [inline src/Index.php:426]: Can you share why this is needed now?

Comment (deguif) [reply src/Index.php:426]: Yes, I had in mind to use option `wait_for_active_shards` when creating indexes on ES.
This would remove the need to have the function `Elastica\Test\Base::_waitForAllocation()` for example by simply passing this option when creating the index.

File: src/Index.php
Comment (ruflin) [inline src/Index.php:406]: If I understand this correctly, it adds an additional call to Elasticsearch each time before an index is created? Assuming index creation is not happening too often, this is probably not a problem but worth mentioning.

File: src/Index.php
Comment (ruflin) [reply src/Index.php:426]: I think I need to read up on my end what this flag does or doesn't. 

Is it now with the new code that `recreate` or was this always the case and now we remove it but didn't before? Trying to understand which part is related to the change of getting the options and what is further improvements.

File: src/Index.php
Comment (deguif) [reply src/Index.php:406]: No extra call to Elasticsearch, just the valid options are retrieved from the underlying endpoint class.

File: src/Index.php
Comment (deguif) [reply src/Index.php:426]: The unset is here only to remove invalid endpoint options so that these options can be passed to the endpoint via `setParams()`.

File: src/Index.php
Comment (ruflin) [reply src/Index.php:406]: Ah, good to know. I somehow assumed it gets it from ES each time.

**Review by ruflin — APPROVED** [review]

**Comment by deguif** [comment]
Waiting for #1867 to be merged before merging this one as I will have to update the tests.

File: tests/IndexTest.php
Comment (deguif) [inline tests/IndexTest.php:586]: Here's a concrete example of the use of native options when creating an index.
This allows to replace the use of `$this->_waitForAllocation()` by a native option.

**Pull request merged**
